ResearchKit/Common/ORKOrderedTask.m

```diff
@@ -353,39 +371,129 @@ + (ORKOrderedTask *)twoFingerTappingIntervalTaskWithIdentifier:(NSString *)ident
             
             ORKStepArrayAddStep(steps, step);
         }
+    }
+    
+    // Setup which hand to start with and how many hands to add based on the handOptions parameter
+    // Hand order is randomly determined.
+    NSUInteger handCount = ((handOptions & ORKPredefinedTaskHandOptionBoth) == ORKPredefinedTaskHandOptionBoth) ? 2 : 1;
+    BOOL undefinedHand = (handOptions == ORKPredefinedTaskHandOptionUndefined);
+    BOOL rightHand;
```

Comment:
After the first pass through, the hand is toggled so if the first pass is `rightHand = YES`, then the second pass is `rightHand = NO`